gips/data/smap/smap.py

```diff
@@ -0,0 +1,221 @@
+        self.date = datetime.datetime.strptime(date_here, "%Y%m%d").date()
+        self._version = (re.search('R[0-9]*', bname)).group(0)
+        self.tile = 'h01v01'
+
+    @classmethod
+    def query_provider(cls, asset, tile, date):
+        """Find out from the SMAP servers what assets are available.
+
+        Uses the given (asset, date) tuple as a search key, andcat
+        returns a tuple:  base-filename, url
+        """
+
+        mainurl = "%s/%s" % (cls._assets[asset]['url'], str(date.strftime('%Y.%m.%d')))
+
+        pattern = r'SMAP\_.{2}\_%s\_%s\_.{6}\_.{3}\.h5' % (asset, str(date.strftime('%Y%m%d')))
+        cpattern = re.compile(pattern)
+        err_msg = "Error downloading: " + mainurl
+        with utils.error_handler(err_msg):
+            response = cls.Repository.managed_request(mainurl, verbosity=2)
+            if response is None:
+                return None, None
+
+        for item in response.readlines():
+            # screen-scrape the content of the page and extract the full name of the needed file
+            # (this step is needed because part of the filename, the creation timestamp, is
+            # effectively random).
+            if cpattern.search(item):
+                if 'xml' in item:
+                    continue
+                basename = cpattern.findall(item)[0]
+                url = ''.join([mainurl, '/', basename])
+                return basename, url
+        utils.verbose_out('Unable to find remote match for '
+                          '{} at {}'.format(pattern, mainurl), 4)
+        return None, None
+
+
+    @classmethod
+    def fetch(cls, asset, tile, date):
+        qs_rv = cls.query_service(asset, tile, date)
+        if qs_rv is None:
+            return []
+        basename, url = qs_rv['basename'], qs_rv['url']
+        with utils.error_handler(
+                "Asset fetch error ({})".format(url), continuable=True):
+            response = cls.Repository.managed_request(url)
+            if response is None:
+                return []
+            outpath = os.path.join(cls.Repository.path('stage'), basename)
+            with open(outpath, 'wb') as fd:
+                fd.write(response.read())
+            utils.verbose_out('Retrieved ' + basename, 2)
+            return [outpath]
+        return []
+
+class smapData(Data):
+    """ A tile of data (all assets and products) """
+    name = 'SMAP'
+    version = '1.0.0'
+    Asset = smapAsset
+
+    _projection = 'PROJCS["unnamed",GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],TOWGS84[0,0,0,0,0,0,0],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9108"]],AUTHORITY["EPSG","4326"]],PROJECTION["Cylindrical_Equal_Area"],PARAMETER["standard_parallel_1",30],PARAMETER["central_meridian",0],PARAMETER["false_easting",0],PARAMETER["false_northing",0],UNIT["Meter",1],AUTHORITY["epsg","6933"]]'
```

Comment:
I know it's a pain but please limit the line width as discussed above.

Reply:
Changed.